### PR TITLE
Jetpack Backup: show progress

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of #15191

This PR shows the backup progress in Activity Log and Backups screen.

#### Creating a backup from the app

1. Go to Activity Log or Backups screen
2. Tap the 3 dots in the first available cell
3. Tap "Download Backup"
4. Tap "Create downloadable file"
5. Tap "Ok notify me"
6. Check that the progress cell is displayed and disappear once the backup is completed

#### Creating a backup from the site

1. Go to your site and trigger a backup there
2. In the app, open Activity Log or Backups Screen
3. The progress bar should be displayed and disappear once the backup is completed

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Stores/ActivityStore.swift [hunk 1]
```diff
@@ -451,6 +540,16 @@ private extension ActivityStore {
...
     }
 
+    func backupStatusUpdated(site: JetpackSiteRef, status: JetpackBackup) {
+        state.backupStatus[site] = status
+
+        if let progress = status.progress, progress > 0 {
+            delayedRetryFetchBackupStatus(site: site)
+        } else {
+            // TODO: show download cell
+        }
```
Comment (emilylaguna) [inline WordPress/Classes/Stores/ActivityStore.swift:550]: TODO coming soon?

File: WordPress/Classes/ViewRelated/Activity/BaseActivityListViewController.swift [hunk 2]
```diff
@@ -499,7 +499,8 @@ extension BaseActivityListViewController: JetpackBackupStatusViewControllerDeleg
             guard let self = self else {
                 return
             }
-            // TODO: fetch backup status
+
+            self.viewModel.refresh()
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Activity/BaseActivityListViewController.swift]: Should we remove the guard and just do:

```suggestion
            self?.viewModel.refresh()
```

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Heya! Looking good, I left some comments below. Also when I start the backup on .com, the once it's finished the status never updates on the app (see video)


https://user-images.githubusercontent.com/793774/106791452-ec0b0b00-6622-11eb-9f75-d23260db779f.mov

File: WordPress/Classes/Stores/ActivityStore.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/Stores/ActivityStore.swift:550]: Yes!

File: WordPress/Classes/ViewRelated/Activity/BaseActivityListViewController.swift (hunk 2 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Activity/BaseActivityListViewController.swift]: Done in ed80398f57db02ca9ca2d3862dbb439b436563ca

**Comment by leandroalonso** [comment]
@emilylaguna I'm not being able to reproduce this behavior here. :( (Atomic site, triggering the backup from the web)

**Review by momo-ozawa — APPROVED** [review]
@leandroalonso @emilylaguna 
I was able to reproduce the error Emily was seeing exactly once, but that's it. I think I saw it on a JN self-hosted site. I tried again with the same site but wasn't able to reproduce the error.

The PR works as described otherwise, so I'm going to approve it. 
I'm thinking we can keep an eye out for when this happens again, then address it in a future PR.

**Review by emilylaguna — APPROVED** [review]
LGTM! 🚀

**Pull request merged**